_posts/2018-06-26-viroreact.md

```diff
@@ -0,0 +1,609 @@
+
+Lights are very important components in a scene.
+Without any light the user wouldn't see anything.
+Luckily a default light is provided when none is defined.
+We didn't need lights in our setup but if you want a more realistic or visually stunning experience, 
+I highly recommend you to look into the different lights in the [documentation](https://docs.viromedia.com/v2.6.1/docs/3d-scene-lighting).  
+There are 4 lighting models: Phong, Blinn, Lambert and Constant.  
+These are the algorithms that calculate how your objects will look like when influenced by light.
+By default, elements use the Constant lighting model, which means lights will be ignored and the object will show its full color.  
+ViroReact supports 4 types of light.
+
+The ViroAmbientLight is the simplest light.
+{% highlight javascript %}
+<ViroAmbientLight color="#ffffff"/>
+{% endhighlight %}
+It lights up all the objects in the scene equally from every direction.
+Only the color needs to be set.
+
+A ViroOmniLight is comparable to a light bulb.
+{% highlight javascript %}
+<ViroOmniLight
+    color="#ffffff"
+    attenuationStartDistance={2}
+    attenuationEndDistance={6}
+    position={[0,-5,5]}
+ />
+{% endhighlight %}
+The light originates from a specified position and moves in all directions.
+The light slowly fades out if the distance is between 2 and 6, set in the attenuation properties.
+
+A ViroSpotLight is comparable to a flashlight.
+{% highlight javascript %}
+<ViroSpotLight
+    color="#ffffff"
+    attenuationStartDistance={2}
+    attenuationEndDistance={6}
+    position={[0, -5, 5]}
+    direction={[0 -1, 0]}
+    innerAngle={0}
+    outerAngle={45}
+ />
+{% endhighlight %}
+The light originates from a single point and shines in a cone.
+The direction and angle of the cone can be set as well as the attenuation start and end distances.
+Some other properties are available to create the shadows you want.
+
+The last type of light is the ViroDirectionalLight.
+{% highlight javascript %}
+<ViroDirectionalLight
+    color="#ffffff"
+    direction={[0, -1, 0]}
+ />
+{% endhighlight %}
+The sun would be the prime example of a ViroDirectionalLight.
+It shines over the whole scene in a specific direction.
+
+#### To make these objects look good we need Materials
+Materials are used to place texture images on 3d objects.
+They can make a long box look like a brick wall or a sphere look like our planet.
+{% highlight javascript %}
+ViroMaterials.createMaterials({
+  earth: {
+	shininess: 2.0,
+	lightingModel: "Lambert",
+	diffuseTexture: require('./res/earth_texture.jpg'),
+  }
+});
+{% endhighlight %}
+There are many properties available for materials.
+Discussing all of these is outside the scope of this article.
+The diffuseColor and diffuseTexture are the main color and texture of the material.
+These can be placed on basic 3D objects like ViroBox, ViroQuad and ViroSphere.
+Complex Viro3DObjects can have multiple materials.
+
+#### 
+// todo: animations
```

Comment:
Todo??

Reply:
Asked michael what still needs to be added, will update later

Reply:
See recent commit 😄